### ISSUE TITLE
Running vet task scans files in vendor directory but I want to validate my own source codes only

### ISSUE THREAD

**Issue by amimas** [gogradle/gogradle#181]
Hi,

I'm trying to incorporate sonarqube analysis to a project that I'm converting to gradle based build using gogradle plugin. Here's the sonarqube plugin for golang I'm trying to use with sonarqube:

https://github.com/uartois/sonar-golang

This tool requires a report generated by `gometaliner`. That's why I'm trying to figure out if gogradle already generates such report by trying out `vet` or `goVet` task. That's why I'm following this example project you've setup to demonstrate `fmt` or `vet` task.

https://github.com/gogradle/examples/tree/master/gofmt-vet-golint

My `build.gradle` looks like this:

```gradle
goVet {
    go 'tool vet -v .'
}
```

When I run `gradlew goVet`, it started to scan everything in my repository but eventually failed with this error:

```
...
vendor/gopkg.in/yaml.v2/yaml.go:167: can't check non-constant format in call to Errorf
Checking file vendor/gopkg.in/yaml.v2/yamlh.go
Checking file vendor/gopkg.in/yaml.v2/yamlprivateh.go

* What went wrong:
Execution failed for task ':golang-project:goVet'.
> Build failed due to return code 1 of: 
  Command:
   /Users/amimas/.gradle/go/binary/1.8.1/go/bin/go tool vet -v .
```
But, I also noticed that the task is scanning the vendor directory as well. My final goal is this:

Run validation on my own source codes (not 3rd party packages in vendor directory). Have this validation available in SonarQube.

Any suggestions? Can gogradle follow gradle convention for project source codes? For example: java plugin expects codes in `src/main/java/` and/or `src/test/java` directory. Also, perhaps the vendor directory can be inside `.gogradle` directory?

**Comment by amimas** [comment]
Would it be possible for gogradle to support gometalinter natively?

https://github.com/alecthomas/gometalinter

I added the following to install `gometalinter` in my gradle project using gogradle

`build.gradle`:
```gradle
task installGoMetalinter(type: com.github.blindpirate.gogradle.Go) {
    dependsOn goVendor //may not be necessary
    go('get -u gopkg.in/alecthomas/gometalinter.v2')
}
```
But I can't seem to execute `gometalinter` on my project. According to the instructions, I need to:
1. Run `gometalinter --install` to install all known linters
2. Run `gometalinter ./...` to execute validation by gometalinter

I tried the following task, but as expected it doesn't work.
`build.gradle`
```gradle
task validate(type: com.github.blindpirate.gogradle.Go) {
    dependsOn installGoMetalinter
    run 'gometalinter --install'
}
```
I get an IOException:
```
* What went wrong:
Execution failed for task ':golang-project:validate'.
> java.io.IOException: Cannot run program "gometalinter" (in directory "/Users/amimas/git-repos/golang-project"): error=2, No such file or directory
```

**Comment by blindpirate** [comment]
Thanks for your suggestion.

For the first issue, I'll consider improving it in next release.

For the second issue, I'd suggest you to install gometalinter into your `PATH` and try:

```
task validate(type: com.github.blindpirate.gogradle.Go) {
    run 'gometalinter --install'
}
```

The `IOException` is due to `gometalinter` doesn't exist in `PATH`.

**Comment by amimas** [comment]
Thanks for your suggestion. Yes, I realized later that I needed to add gometalinter to the `PATH`. 

I hope you'll also consider adding gometalinter to gogradle as it includes a lot of linters already including `go vet`. Or, perhaps there can be an additional configuration in gogradle for users to run their own/custom tools.

**Comment by blindpirate** [comment]
Closed by `0.9`. Thanks for reporting.
